### DISCUSSION TITLE
Is this adapted to use in rails engines?

### DISCUSSION THREAD

**Discussion by irakliasatiani** [rswag/rswag#615] [General]
I have lot's of engines that act like services, each one should have its own swagger, the problem is that when I try to have separated endpoints, they don't find swagger.yaml files for each and shows 404 error, when I don't change endpoint however it works but they show the same swagger file which makes sense, is it possible to use it in engines? and if yes, what's appropriate configuration, say to configure it for two engines which in turn are being connected to host app.

**Comment by duychaufit** [comment]
@buxbunny110 
did you try to config the swagger_endpoint for each of your engines ?
 
`config/initializers/rswag_ui.rb`
```
Rswag::Ui.configure do |c|
  swagger_endpoint = condition ? '/api-docs/v1/engine1.yaml' : '/api-docs/v1/engine2.yaml'
  c.swagger_endpoint swagger_endpoint, 'Swagger API V1 Docs'
end
```

**Reply by irakliasatiani** [reply]
Yes, I tried to change endpoint for each engine but I have feeling like all of them are picking only the one of them, that been specified
